### ISSUE TITLE
Settings not saved on Windows and Android since Upgrade to v1.0.8 and v1.0.9

### ISSUE THREAD

**Issue by Aquariusquintini** [AndreasReitberger/MauiSettings#56]
I upgraded the used version of MauiSettings in my project from v1.0.7 to v1.0.9 and noticed that settings are not saved anymore between runs of my program. It seems that AppSettings.SaveSettings(); silently fails, no errors what so ever. 

```cs
 // Save that we succesfully downloaded the resources for the current version
 AppSettings.ResourcesCurrentVersionAvailable = AppInfo.Current.VersionString;
 AppSettings.SaveSettings();`

 public partial class AppSettings : MauiSettings<AppSettings>
 {

     #region Settings

     #region Version
     [MauiSetting(Name = nameof(ResourcesCurrentVersionAvailable))]
     public static string ResourcesCurrentVersionAvailable { get; set; }
```

Steps to reproduce:
* Run Program
* Read a setting
* Change its value and save the changes
* Restart the program

Expected behavior: the value read is 'NewValue' actual 'null'.

Any help on this is highly appreciated.

**Comment by AndreasReitberger** [comment]
@Aquariusquintini 
You're welcome. I created a small test page in the `SampleApp` and tried to reproduce the issue. On my end, it is working fine however.

![image](https://github.com/user-attachments/assets/dac97eed-3545-439b-8de8-5c568542c3b2)
https://github.com/AndreasReitberger/MauiSettings/blob/56-settings-not-saved-on-windows-and-android-since-upgrade-to-v108-and-v109/src/MauiSettings.Example/ViewModels/SettingsPageViewModel.cs

Can you create a small sample repro showing this issue? I'll investigate it then.

Thank you!

**Comment by Aquariusquintini** [comment]
Dear Andreas,

Thank you for the quick replies. 

I tried to investigate the issue further and tried to create a small repro. However, in my repro-project everything is working fine. In my actuall app, it fails consistently, after LoadSettings the value = "null" instead of the value saved. I tried to clear the preferences and uninstall the app, but that did not help. I also tried to create the settings using v1.0.7 and then switch to 1.0.9, that also worked fine.

I will try to investigate it further and get back to you as soon as I have more information.

Kind regards

**Comment by AndreasReitberger** [comment]
Hi @Aquariusquintini 
I found a small bug when saving numeric values. This was fixed by #62.
If your issue still persist, please try to share a small repro project. Till that, I will close this issue. Feel free to re-open a new one if you still have troubles.

Thanks!

**Comment by Aquariusquintini** [comment]
I can confirm that this fixed the problem. Thank you very much!
